File: javaa/Dokter.java
package javaa;

public class Dokter extends Manusia {
    private String str; //Surat Tanda Registrasi (No.izin)
    private long gaji_pokok; //tidak boleh sembarangan dirubah jadi private
    private String rumah_sakit; 
    private static int str_counter = 1000; //ini untuk str unik yang tidak bisa diakses siapapun sembarangan
    //harus berupa static agar atribut dianggap milik class dan setiap kali ada objek baru nilainya akan meningkat

    public Dokter(String nama, int umur, String gender, long gaji_pokok, String rumah_sakit) {
        super(nama, umur, gender); //di java wajib inisialisasi constructor parent dengan sintaks super (ini constructor Manusia)
        setGajiPokok(gaji_pokok);
        setRumahSakit(rumah_sakit);
        this.str = generateStrOtomated(); //nantinya teks akan masuk ke str
    }

    private String generateStrOtomated() { //Surat tanda registrasi tidak bisa sembarangan diubah oleh siapapun
        str_counter++; //inin akan mengcounter nilai milik class
        return "STR-" + str_counter; 
    }

    //getter
    public String getStr() {
        return str;
    }
    public String getRumahSakit() {
        return rumah_sakit;
    }
    public long getGajiPokok() {
        return gaji_pokok;
    }

    //setter
    public void setRumahSakit(String rumah_sakit) {
        this.rumah_sakit = rumah_sakit;
    }
    public void setGajiPokok(long gaji_pokok) {
        this.gaji_pokok = gaji_pokok;
    }

    //informasi
    public void informasiDokter() {
        System.out.println(getNama());
        System.out.println(getUmur());
        System.out.println(getGender());
        System.out.println(getStr());
        System.out.println(getGajiPokok());
        System.out.println(getRumahSakit());
    }
}

File: pythonn/Dokter.py
from Manusia import Manusia

class Dokter(Manusia):
    _nomor_regis_counter = 1000 #static protected, ga pake private nanti pusing namling 

    def __init__(self, nama, umur, gender, gaji_pokok, rumah_sakit) :
        #isi construct parent class
        super().__init__(nama, umur, gender) 
        self.setGajiPokok(gaji_pokok)
        self.setRumahSakit(rumah_sakit)

        self.__nomor_regis = self.generateStrOtomatis()

    def generateStrOtomatis(self) :
        #Dokter karena static milik class, bukan instance
        Dokter._nomor_regis_counter += 1
        return "STR-" + str(Dokter._nomor_regis_counter) #str mengubah int ke string 

    #setter
    def setGajiPokok(self, gaji_pokok) :
        if (gaji_pokok <= 0) :
            print("Harus punya gaji")
            return
        self.__gaji_pokok = gaji_pokok

    def setRumahSakit(self, rumah_sakit) :
        self.__rumah_sakit = rumah_sakit

    #getter
    def getNoRegis(self) :
        return self.__nomor_regis
    
    def getGajiPokok(self) :
        return self.__gaji_pokok

    def getRumahSakit(self) :
        return self.__rumah_sakit

    def informasiDokter(self) :
        print(self.getNoRegis())
        print(self.getNama())
        print(self.getUmur())
        print(self.getGender())
        print(self.getGajiPokok())
        print(self.getRumahSakit())
